### PR TITLE
#M20. Add automated emails to signup and hackathon registration

### PR DESCRIPTION
**Pull request by stefdworschak**
## Description
- Add email templates to be sent on enrolment / withdrawal

## Pull request type

- [x] #M01 (Miscellaneous User Story #01)
- [ ] #P02 (Participant User Story #02)
- [ ] #S03 (Staff User Story #03)
- [ ] #A04 (Admin User Story #04)
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue
#193

## Configuration instructions
- This feature needs to new services: 1 Redis & 1 celery worker
- Also needs a functioning SMTP Server for production

### CODE REVIEW

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<p>
+    <strong>What does participation involve?</strong><br>
+    You'll be assigned to a team, and work together building a project based on the assigned theme in a limited number of days. Don't worry if you have limited coding experience, all levels are welcome, and we encourage alumni to participate!
+</p>
+
+<p>
+    <strong>What am I committing to?</strong><br>
+    We recommend at bare minimum you dedicate a minimum of 8-10 hours over the duration of the Hackathon. You will be expected to actively contribute to your team, not just observe.
+    Please check your calendar and confirm that you are available before registering as dropping out really lets your team down and the team will be a person short.
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:13]: Grammar suggestions:

Please check your calendar and confirm that you are available before registering`,` as dropping out really lets your team down`,` and the team will be `one` person `fewer`.

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+<p>
+    <strong>What am I committing to?</strong><br>
+    We recommend at bare minimum you dedicate a minimum of 8-10 hours over the duration of the Hackathon. You will be expected to actively contribute to your team, not just observe.
+    Please check your calendar and confirm that you are available before registering as dropping out really lets your team down and the team will be a person short. 
+</p>
+
+<p>
+    <strong>!IMPORTANT</strong><br>
+    Please ensure your Profile is up to date, especially the 'Latest Module' entry. This is vital for the team selection process. We try our best to balance the teams fairly so it really helps you and your team to be accurate with your profile.
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:18]: Grammar suggestions:

Please ensure your Profile is up to date, especially the 'Latest Module' entry. This is vital for the team selection process. We try our best to balance the teams fairly`,` so it really helps you and your team to be accurate with your profile.

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+</p>
+
+<p>
+    <strong>What am I committing to?</strong><br>
+    We recommend at bare minimum you dedicate a minimum of 8-10 hours over the duration of the Hackathon. You will be expected to actively contribute to your team, not just observe.
+    Please check your calendar and confirm that you are available before registering as dropping out really lets your team down and the team will be a person short. 
+</p>
+
+<p>
+    <strong>!IMPORTANT</strong><br>
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:17]: Exclamation meant to be at the start?
Perhaps:

`<strong>IMPORTANT!</strong><br>`

This matches the next paragraph

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    Please ensure your Profile is up to date, especially the 'Latest Module' entry. This is vital for the team selection process. We try our best to balance the teams fairly so it really helps you and your team to be accurate with your profile. 
+</p>
+
+<p>
+    <strong>Register for the Intro Webinar!</strong><br>
+    Please check the #hackathon channel for details on how to register for the intro and project presentations webinar.
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:23]: Link consideration for directing to the Hackathon channel:

```
Please check the <a href="https://code-institute-room.slack.com/archives/CDAFARB71" target="_blank">#hackathon</a> channel for details on how to register for the intro and project presentations webinar.
```

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+</p>
+
+<p>
+    <strong>Register for the Intro Webinar!</strong><br>
+    Please check the #hackathon channel for details on how to register for the intro and project presentations webinar. 
+</p>
+
+<p>
+    <strong>Need Help?</strong><br>
+    Please ask any questions in the #hackathon channel, the HackTeam are ready and happy to help out. You can ask them a question by using the @hackteam tag on slack.
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:28]: Same here:

```
Please ask any questions in the <a href="https://code-institute-room.slack.com/archives/CDAFARB71" target="_blank">#hackathon</a> channel; the HackTeam are ready and happy to help out. You can ask them a question by using the @hackteam tag on slack.
```

File: hackathon/templates/hackathon/enrolment_email.txt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    <strong>Register for the Intro Webinar!</strong><br>
+    Please check the #hackathon channel for details on how to register for the intro and project presentations webinar. 
+</p>
+
+<p>
+    <strong>Need Help?</strong><br>
+    Please ask any questions in the #hackathon channel, the HackTeam are ready and happy to help out. You can ask them a question by using the @hackteam tag on slack. 
+</p>
+
+<p>Thanks again for signing up, we are excited to see what you and your team will create! Remember, hackathons are about team-building, learning and most importantly having fun.
```
Comment (TravelTimN) [inline hackathon/templates/hackathon/enrolment_email.txt:31]: Grammar:

Thanks again for signing up`,` we are excited to see what you and your team will create! Remember, hackathons are about team-building, learning`,` and most importantly having fun.

**Review by TravelTimN — APPROVED** [review]
Minor suggestions, but otherwise LGTM 👍🏻

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 1 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 2 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 3 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 4 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 5 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

File: hackathon/templates/hackathon/enrolment_email.txt (hunk 6 above)
Comment (stefdworschak) [reply hackathon/templates/hackathon/enrolment_email.txt]: Updated as suggested

**Pull request merged**